docs/design/datamover.md

```diff
@@ -0,0 +1,239 @@
+---
+title: datamover_crd_design
+authors:
+  - "@savitharaghunathan"
+reviewers:
+  - "@shawn_hurley"
+  - "@alaypatel07"
+  - "@dymurray"
+  - "@eemcmullan"
+approvers:
+  - "@shawn_hurley"
+  - "@alaypatel07"
+creation-date: 2022-03-16
+status: provisional
+---
+
+# Data Mover CRD design
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Open questions
+
+* PVC mover/ volumesnapshot mover
+    * Talk through how are we going to support these two types
+    * Snapshots as de-facto standard in the Kube world
+* Tech preview conversation
+    * Upstream vs downstream
+
+## Summary
+OADP operator currently supports backup and restore of applications backed by CSI volumes. It takes advantage of Velero CSI plugin to achieve the same. However, these snapshots are local to the Openshift cluster and cannot be recovered if the cluster gets deleted accidentally or if there is a disaster. In order to overcome this issue, DataMover is made available for users to save the snapshots in a remote storage. 
+
+## Motivation
+
+Create an extensible design to support various data movers that can be integrated with OADP operator. Vendors should be able to bring their own data mover controller and implementation, and use that with OADP operator.
+
+## Goals
+* Create an extensible data mover solution
+* Supply a default data mover option 
+* Supply APIs for DataMover CRs (eg: DataMoverBackup, DataMoverRestore,DataMoverType)
+* Supply a sample codebase for the Data Mover plugin and controller implementation
+
+
+## Non Goals
+* Maintain 3rd party data mover implementations
+* Adding a status watch controller to Velero
+
+## User stories
+
+Story 1: 
+As an application developer, I would like to save the CSI snaphots in a S3 bucket. 
+
+Story 2:
+As a cluster admin, I would like to be able to restore CSI snapshots if disaster happens.
+
+## Design & Implementation details
+
+This design supports adding the data mover feature to the OADP operator and facilitates integrating various vendor implemented data movers. 
+
+![DataMover CRD](../images/datamovercrd.png)
```

Comment:
Please update this image path.

Reply:
Should I create a PR to just merge the image first? I think the path is correct. WDYT?

Reply:
Looks fine to me @ https://github.com/savitharaghunathan/oadp-operator/blob/data_mover_enhancement/docs/design/datamover.md

GitHub PR .md previews don't show (new??) images.

Reply:
@kaovilai I think so.